nearcore/src/config.rs

```diff
@@ -577,15 +577,23 @@ impl NearConfig {
                 network_key_pair.secret_key,
                 validator_signer.clone(),
                 config.archive,
-            ),
+                // enable_accounts_data:
+                match genesis.config.chain_id.as_ref() {
+                    // disable in mainnet, testnet and betanet.
+                    "mainnet" | "testnet" | "betanet" => false,
+                    // enable in shardnet and all test setups.
+                    "shardnet" => true,
+                    _ => true,
+                },
```

Comment:
cc @mm-near, this is how we are going to gate tier1 net. Calling to attention because this is different to how we traditionally approach feature gating (runtime config, rather than a compile time flag). I fully support this approach :)